internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -424,17 +424,23 @@ func reconcileExternalTemplateReference(ctx context.Context, c client.Client, cl
 		return err
 	}
 
+	desiredOwnerRef := metav1.OwnerReference{
+		APIVersion: clusterv1.GroupVersion.String(),
+		Kind:       "Cluster",
+		Name:       cluster.Name,
+		UID:        cluster.UID,
+	}
+
+	if util.HasExactOwnerRef(obj.GetOwnerReferences(), desiredOwnerRef) {
```

Comment:
Note: this here lacks checking if the uid matches :think:

Reply:
(same in machineset controller)

Reply:
Yup. I think you're right. I'll follow-up tomorrow with a PR against main

Reply:
Let's merge this PR independent of that. Will open a PR against main in a few minutes and then we can cherry-pick that one into 1.9 + 1.8

Reply:
Fix PR: https://github.com/kubernetes-sigs/cluster-api/pull/11688